### PR TITLE
update types of @hippy/react

### PR DESCRIPTION
**Pull request by zerosrat**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://hippyjs.org/#/hippy-react/components)>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## hippy__react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/hippy__react@latest/))
was missing the following properties:
1. Animated
1. AppRegistry
1. Clipboard
1. ConsoleModule
1. Easing

<details>
<summary>as well as these 16 other properties...</summary>
<p>Focusable, HippyRegister, ImageBackground, ImageLoaderModule, ListViewItem, NetworkModule, PixelRatio, RNfqb, RNfqbEventEmitter, RNfqbEventListener, RNfqbRegister, TimerModule, WebSocket, WebView, colorParse, flushSync</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e6a4354208ded4280f570e14e242378dd6e73564
</p>

File: types/hippy__react/index.d.ts [hunk 1]
```diff
@@ -935,7 +940,7 @@ interface TextProps extends LayoutableProps, ClickableProps, TouchableProps {
      * The default is `tail`.
      */
     ellipsizeMode?: 'head' | 'middle' | 'tail' | 'clip';
-    children?: number | string | string[];
+    children?: number | string | string[] | React.ReactNode;
```
Comment (martin-badin) [inline types/hippy__react/index.d.ts:943]: `ReactNode` can be `string | number` based on the source file
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L232-L240
```suggestion
    children?: string[] | React.ReactNode;
```

File: types/hippy__react/index.d.ts [hunk 2]
```diff
@@ -1496,5 +1724,41 @@ interface UIManagerModule {
...
+     * The callback after rendering.
+     */
+    callback?: () => void | undefined | null;
+}
+declare class Hippy {
+    constructor(config: HippyReactConfig);
+    config: HippyReactConfig;
+    rootContainer: any;
+    // Keep forward comaptatble.
+    regist: () => void;
```
Comment (martin-badin) [inline types/hippy__react/index.d.ts:1759]:
```suggestion
    regist(): void;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

**Review by martin-badin — APPROVED** [review]

**Comment by zerosrat** [comment]
Ready to merge

**Pull request merged**